src/accumulator/mod.rs

```diff
@@ -1,3 +1,67 @@
+//! This module is the core of the library. It contains all the basic data structures to use Utreexo,
+//! such as accumulator, and proofs. Even though the algorithms are equal, you may use different
+//! data structures to represent the accumulator, each with its own advantages and disadvantages.
+//!
+//! # [Stump]
+//! A Stump is a basic data structure used in Utreexo. It only holds the roots and the number of leaves
+//! in the accumulator. This is useful to create lightweight nodes, the still validates, but is more compact,
+//! perfect to clients running on low-power devices.
+//!
+//! ## Example
+//! ```
+//!   use rustreexo::accumulator::{types::NodeHash, stump::Stump, proof::Proof};
+//!   use std::str::FromStr;
+//!   // Create a new empty Stump
+//!   let s = Stump::new();
+//!   // The newly create outputs
+//!   let utxos = vec![NodeHash::from_str("b151a956139bb821d4effa34ea95c17560e0135d1e4661fc23cedc3af49dac42").unwrap()];
+//!   // The spent outputs
+//!   let stxos = vec![];
+//!   // Modify the Stump, adding the new outputs and removing the spent ones, notice how
+//!   // it returns a new Stump, instead of modifying the old one. This is due to the fact
+//!   // that modify is a pure function, and none of it's operands are modified in the process.
+//!   let s = s.modify(&utxos, &stxos, &Proof::default());
+//!   assert!(s.is_ok());
+//!   assert_eq!(s.unwrap().0.roots, utxos);
+//! ```
+//!
+//! # Proof
+//! A proof is a data structure that proves that a given element is in the accumulator. It is used
+//! to validate a transaction. A proof is composed of a list of hashes and a list of integers. The
+//! hashes are the siblings of the element in the accumulator. The integers are the position of the
```

Comment:
`s/hashes are the siblings of the element in the accumulator./hashes are the hashes need to calculate the root hash for validation./`

Something like that would be better since it can be the aunt of the element being proven.